ext/MitsubaLoader/CMitsubaLoader.h

```diff
@@ -49,269 +49,165 @@ namespace bsdf
 	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_OPACITY_TEX = 10u;
 	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_MASK_PHONG_EXP_TEX = 0x1u;
 	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_PHONG_EXP_TEX = 16u;
-	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_MASK_MIXTURE_CHILD_COUNT = 0x1fu;
-	_IRR_STATIC_INLINE_CONSTEXPR uint32_t BITFIELDS_SHIFT_MIXTURE_CHILD_COUNT = 10u;
+
+	//padding which make sure struct has sizeof at least min_desired_size but aligned to 4*sizeof(uint32_t)=16
+	//both size and desired_size are meant to be expressed in 4 byte units (i.e. to express size of 8, `size` should be 2)
+	template<size_t size, size_t min_desired_size, size_t diff = (min_desired_size-size)>
+	using padding_t = uint32_t[ ((min_desired_size + 3ull) & ~(3ull)) - size ];
+	//in 4 byte units
+	constexpr size_t max_bsdf_struct_size = 10ull;
+
 #include "irr/irrpack.h"
-	/**
-	GLSL:
-	struct {
-		uint bitfields;
-		vec2 alpha;
-		vec3 reflectance;
-	};
-	*/
 	struct alignas(16) SAllDiffuse
 	{
 		uint32_t bitfields;
-		float dummy;
 		//if flag decides to use alpha texture, {alpha[0..1]} is bindless texture ID (bit-cast to uvec2)
 		//otherwise alpha[0] is single-float alpha
-		float alpha[2];
-		//if flag decides to use reflectance texture, {reflectance[0..1]} is bindless texture ID (bit-cast to uvec2)
-		//otherwise reflectance[0..2] is constant reflectance
-		float reflectance[3];
+		uint32_t alpha[2];
+		//if flag decides to use reflectance texture, `reflectance` is bindless texture ID (bit-cast to uvec2)
+		//otherwise `reflectance` is constant reflectance in RGB19E7 format
+		uint64_t reflectance;
+		padding_t<5, max_bsdf_struct_size> padding;
 	} PACK_STRUCT;
-	/**
-	GLSL:
-	struct {
-		uint bitfields;
-		vec3 transmittance;
-	};
-	*/
 	struct alignas(16) SDiffuseTransmitter
 	{
 		uint32_t bitfields;
-		float dummy[3];
-		//if flag decides to use transmittance texture, {transmittance[0..1]} is bindless texture ID (bit-cast to uvec2)
-		//otherwise transmittance[0..2] is constant transmittance
-		float transmittance[3];
+		//if flag decides to use transmittance texture, `transmittance` is bindless texture ID (bit-cast to uvec2)
+		//otherwise `transmittance` is constant transmittance
+		uint64_t transmittance;
```

Comment:
you need a single dummy, because `vec2` won't go on a 4byte offset, only 8 byte or 16 byte